app/Http/Controllers/System/RoomController.php

```diff
@@ -39,6 +40,7 @@ public function index()
             'roles' => Role::all(),
             'availableRoomTypes'=> config(self::ROOM_TYPES),
             'availableBuildings' => config(self::BUILDING_NAMES),
+            'paginator' => Room::with('restrictions', 'availabilities', 'blackouts', 'dateRestrictions')->paginate(self::ROOM_PAGINATOR_AMOUNT)
```

Comment:
>Paginator holds everything you need for rooms. they both hold teh same data so this is making the same call twice. 
If you do not want to change to much i would suggest only returning the pagination data as rooms. Then in the vue, when you pass rooms into the table pass paginator.data since they are the same. 
Same thing here